### PR TITLE
Fixes UI race conditions, and adds a permanent http client

### PR DESCRIPTION
**Pull request by squizzling**
tcell 2.2.0 contains a bug with exiting that causes a deadlock in the UI
when run under Windows.  This was fixed in 2.2.1, however there is also a
race condition which is fixed in 2.3.1.  This also commits go.sum, which
should be present.  There is no guarantee that a dependency won't be
compromised in future, so this will lock everything to explicit commits.

This also splits how data is retrieved.  You're only meant to edit widgets
from the main tview loop.  This does the http access on the worker routine,
then pushes the updates in to the tview loop.  QueueUpdateDraw is sync
and will wait for the function to complete before resuming.

Finally we add a top level http.Client.  Creating a new client every time
will stop connection re-use.

### CODE REVIEW

File: internal/client.go [hunk 1]
```diff
@@ -58,13 +63,13 @@ func (client *Client) checkServers() {
 	}
 }
 
-func (client *Client) checkServer(host string) {
-	c := &http.Client{}
+func (client *Client) getServerData(host string) (*Msg, error) {
 	req, err := http.NewRequest("GET", fmt.Sprintf("http://%s/", host), nil)
 	if err != nil {
-		client.logTextbox.SetText(err.Error())
+		return nil, err
```
Comment (squizzling) [inline internal/client.go:69]: This also fixes a subtle bug where is http.NewRequest failed, we would continue executing, resulting in more errors down below.  In practice that's not a real problem, but something to be aware of.

**Comment by maded2** [comment]
'
Finally we add a top level http.Client. Creating a new client every time
will stop connection re-use.
'

Actually, since I added multiple plotter support.  I dont want to re-use the same http client.  Can you remove this part as I am fine with the rest of the PR.   Thanks for your contributions.

**Comment by squizzling** [comment]
I could, but what's the concern with sharing a client? A client basically provides settings and connection reuse / pooling.

**Comment by maded2** [comment]
actually, you are correct.  Let's keep the global http client.

**Comment by squizzling** [comment]
Groovy. Please merge, and I'll start finishing up the next PR.

**Pull request merged**